Feature/cdo results count

elizabeth-jimenez opened the pull request:
Acceptance Criteria:

- [ ] Default number of clients should be 10:
- [ ] 10 in dropdown
- [ ] 10 results shown
- [ ] When all is selected all of the clients show?
- [ ] Page resets to 1 when number of clients changes.
- [ ] All of the above with one CDO and > 1CDO

mjoyce91 commented on src/Containers/BidderPortfolio/BidderPortfolio.jsx:
```diff
@@ -60,18 +59,25 @@ class BidderPortfolio extends Component {
+    let qmod;
+    // just in case the dropdown did not get set
+    if (isEqual(q, { limit: '0' })) {
+      qmod = { limit: get(this.props.bidderPortfolio, 'count') };
```
I don't think you want to make use of the count to set the limit for "all". The reason being, I believe it would fail in this scenario:

1. Clients are fetched - there are 400 clients
2. User changes page size dropdown to "all"
3. User decides to change the filter to "handshake only" and gets 250 results
4. User unsets the "handshake only" fillter - but the 250 count from `get(this.props.bidderPortfolio, 'count')` persists and is passed to a query that actually has 400 results

I would look into a way to do this server-side

elizabeth-jimenez replied:
ah, good catch, will do!

mjoyce91 replied:
Maybe a custom "all" value that just maps to `99999`

elizabeth-jimenez replied:
hehe that's the exact format i had before

mjoyce91 commented:
Let's see if we can fix this. May need to work with Sarah

<img width="1189" alt="Screen Shot 2020-03-19 at 1 53 59 PM" src="https://user-images.githubusercontent.com/11576347/77098698-265ac200-69e9-11ea-9213-201fba1b236f.png">

mjoyce91 commented on src/Constants/Sort.js:
```diff
@@ -54,6 +54,18 @@ export const POSITION_PAGE_SIZES = {
+    { value: 10, text: '10' },
+    { value: 25, text: '25' },
+    { value: 50, text: '50' },
+    { value: 0, text: 'all' },
```
I think we do `'All'` elsewhere

mjoyce91 commented on src/Components/BidderPortfolio/BidControls/BidControls.jsx:
```diff
@@ -36,9 +37,17 @@ class BidControls extends Component {
+      defaultOrdering, pageSize, totalClients } = this.props;
     const { hasSeasons } = this.state;
+    const pageSizes = CLIENTS_PAGE_SIZES.options;
+    pageSizes[4].value = totalClients;
```
I would use something more reliable than the index (let's say we add a new option to the beginning of the array in the future). Maybe do a `find` for the object with the value you want.

elizabeth-jimenez replied:
sounds good!

elizabeth-jimenez commented:
> Let's see if we can fix this. May need to work with Sarah
> 
> <img alt="Screen Shot 2020-03-19 at 1 53 59 PM" width="1189" src="https://user-images.githubusercontent.com/11576347/77098698-265ac200-69e9-11ea-9213-201fba1b236f.png">

![image](https://user-images.githubusercontent.com/23065001/77100070-63c04f00-69eb-11ea-880a-32441f6baf95.png)
How about stacking them on smaller screens?

mjoyce91 commented:
@elizabeth-jimenez that looks much better!

elizabeth-jimenez commented:
> Let's see if we can fix this. May need to work with Sarah
> 
> <img alt="Screen Shot 2020-03-19 at 1 53 59 PM" width="1189" src="https://user-images.githubusercontent.com/11576347/77098698-265ac200-69e9-11ea-9213-201fba1b236f.png">

Updated filters tested on resolutions: 1700, 1500, 1300, 1200, 900, 500. Looks good on all.

mjoyce91 commented:
@elizabeth-jimenez - this conditional filter becomes a little weird with the update. Maybe "Filter by" should become the later filter since it's displayed conditionally. Or it could be grayed out until a season is chosen.

![Mar-23-2020 11-49-26](https://user-images.githubusercontent.com/11576347/77335458-8eb4e680-6cfc-11ea-85cb-455a461c2745.gif)

elizabeth-jimenez commented:
> @elizabeth-jimenez - this conditional filter becomes a little weird with the update. Maybe "Filter by" should become the later filter since it's displayed conditionally. Or it could be grayed out until a season is chosen.
> 
> ![Mar-23-2020 11-49-26](https://user-images.githubusercontent.com/11576347/77335458-8eb4e680-6cfc-11ea-85cb-455a461c2745.gif)

Updated! Disabled when Bid Season not chosen and also moved to be after Bid Season filter, bc it makes more sense.

mjoyce91 commented on src/Components/BidderPortfolio/BidControls/BidControls.jsx:
```diff
@@ -47,22 +55,29 @@ class BidControls extends Component {
+                  label="Filter by:"
+                  defaultSort={defaultHandshake}
+                  disabled={!hasSeasons}
+                />
```
Can we gray out this form if it's disabled? Otherwise its not clear why the user is not able to select it

mjoyce91 commented:
I like the spacing, but I wonder if we'd want to push the dropdowns on the bottom (but not their labels) to the right so that they're flush with the ones on the top?

<img width="621" alt="Screen Shot 2020-03-25 at 12 42 38 PM" src="https://user-images.githubusercontent.com/11576347/77562286-33b6f700-6e96-11ea-8d59-7aa67a1740f4.png">

elizabeth-jimenez commented:
> I like the spacing, but I wonder if we'd want to push the dropdowns on the bottom (but not their labels) to the right so that they're flush with the ones on the top?
> 
> <img alt="Screen Shot 2020-03-25 at 12 42 38 PM" width="621" src="https://user-images.githubusercontent.com/11576347/77562286-33b6f700-6e96-11ea-8d59-7aa67a1740f4.png">

I considered that too, but thought it looked a little strange. What do you think?
![image](https://user-images.githubusercontent.com/23065001/77563247-64e3f700-6e97-11ea-8041-55ca9c986f7a.png)

elizabeth-jimenez commented:
🙏 🤞
